addon/services/can.js

```diff
@@ -10,16 +11,16 @@ export default Service.extend({
     return normalizeAbilityString(abilityString);
   },
 
+  abilityFor(abilityName, model, properties = {}) {
+    let Ability = getOwner(this).factoryFor(`ability:${abilityName}`);
 
-    return this.abilityFor(...arguments);
-  },
+    assert(`No ability type found for '${abilityName}'`, Ability);
 
-  abilityFor(abilityName, model, properties = {}) {
-    let ability = getOwner(this).lookup(`ability:${abilityName}`);
-    assert(`No ability type found for ${abilityName}`, ability);
+    if (typeof model != 'undefined') {
+      properties = assign({}, { model }, properties);
+    }
 
-    ability.setProperties(assign({}, { model }, properties));
-    return ability;
+    return Ability.create(properties);
```

Comment:
The `can` method calls `valueFor` which is doing this. Does this comment is meant to be to an outdated code line?

Reply:
Ah, sorry.. I clicked on the commit ("Instantiate abilities") to check how this is going to be handled, since we are currently running into exactly the problem this is going to solve - and there I saw this. Looks great in the actual final code, so please ignore this - thanks for working on this!